libp2p/protocols/kademlia/kademlia.nim

```diff
@@ -269,29 +262,38 @@ proc bootstrap*(
     try:
       await kad.switch.connect(b.peerId, b.addrs)
       debug "connected to bootstrap peer", peerId = b.peerId
-    except CatchableError as e:
-      error "failed to connect to bootstrap peer", peerId = b.peerId, error = e.msg
-
-    try:
-      let msg = await kad.sendFindNode(b.peerId, b.addrs, kad.rtable.selfId).wait(
-        chronos.seconds(5)
-      )
-      for peer in msg.closerPeers:
-        let p = PeerId.init(peer.id).tryGet()
-        discard kad.rtable.insert(p)
+    except DialFailedError as e:
+      # at some point will want to bubble up a Result[void, SomeErrorEnum]
+      error "failed to dial to bootstrap peer", peerId = b.peerId, error = e.msg
+      continue
+
+    let msg =
+      try:
+        await kad.sendFindNode(b.peerId, b.addrs, kad.rtable.selfId).wait(
+          chronos.seconds(5)
+        )
+      except CatchableError as e:
+        debug "send find node exception during bootstrap",
+          target = b.peerId, addrs = b.addrs, err = e.msg
+        continue
```

Comment:
Do you need this continue?

Reply:
it's either continue or return. it makes no sense to progress for this particular bootstrap node, which in itself implies a return is needed, but there may be other bootstrap nodes to try, which means instead of return, a continue is needed.

the rest of the loop is dependant on the `msg` reply being valid.